aiohttp_sse/__init__.py

```diff
@@ -149,49 +166,86 @@ def last_event_id(self) -> Optional[str]:
 
         return self._req.headers.get(self.DEFAULT_LAST_EVENT_HEADER)
 
+    @property
+    def ping_interval(self) -> Union[int, float]:
+        """Time interval between two ping massages"""
+        return self._ping_interval
+
     @ping_interval.setter
-    def ping_interval(self, value):
+    def ping_interval(self, value: Union[int, float]) -> None:
         """Setter for ping_interval property.
 
         :param int value: interval in sec between two ping values.
         """
 
-        if not isinstance(value, int):
-            raise TypeError("ping interval must be int")
+        if not isinstance(value, (int, float)):
+            raise TypeError("ping interval must be int or float")
         if value < 0:
             raise ValueError("ping interval must be greater then 0")
 
         self._ping_interval = value
 
-    async def _ping(self):
+    async def _ping(self) -> None:
         # periodically send ping to the browser. Any message that
         # starts with ":" colon ignored by a browser and could be used
         # as ping message.
         while True:
             await asyncio.sleep(self._ping_interval)
             try:
                 await self.write(": ping{0}{0}".format(self._sep).encode("utf-8"))
-            except ConnectionResetError:
-                self._ping_task.cancel()
+            except ConnectionResetError as exc:
+                raise asyncio.CancelledError() from exc
 
-    async def __aenter__(self):
+    async def __aenter__(self) -> "EventSourceResponse":
         return self
 
-    async def __aexit__(self, *args):
+    async def __aexit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc: Optional[BaseException],
+        traceback: Optional[TracebackType],
+    ) -> None:
         self.stop_streaming()
         await self.wait()
-        return
+
+
+ESR = TypeVar("ESR", bound=EventSourceResponse)
+
+
+@overload
+def sse_response(
+    request: Request,
+    *,
+    status: int = 200,
+    reason: Optional[str] = None,
+    headers: Optional[Mapping[str, str]] = None,
+    sep: Optional[str] = None,
+) -> _ContextManager[EventSourceResponse]:
+    ...
+
+
+@overload
+def sse_response(
+    request: Request,
+    *,
+    status: int = 200,
+    reason: Optional[str] = None,
+    headers: Optional[Mapping[str, str]] = None,
+    sep: Optional[str] = None,
+    response_cls: Type[ESR] = ...,
```

Comment:
With the default, it overlaps with the first overload.
```suggestion
    response_cls: Type[ESR],
```